internal/handler/wallet/create_test.go

```diff
@@ -0,0 +1,106 @@
+package wallet
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/go-openapi/loads"
+	"github.com/massalabs/thyra-plugin-massa-wallet/api/server/models"
+	"github.com/massalabs/thyra-plugin-massa-wallet/api/server/restapi"
+	"github.com/massalabs/thyra-plugin-massa-wallet/api/server/restapi/operations"
+)
+
+func configureAPIServer() (*operations.MassaWalletAPI, error) {
+	swaggerSpec, err := loads.Analyzed(restapi.SwaggerJSON, "")
+	if err != nil {
+		return nil, err
+	}
+
+	localAPI := operations.NewMassaWalletAPI(swaggerSpec)
+	server := restapi.NewServer(localAPI)
+
+	var walletStorage sync.Map
+
+	localAPI.RestWalletCreateHandler = NewCreate(&walletStorage)
+
+	server.ConfigureAPI()
+
+	return localAPI, nil
+}
+
+func Test_walletCreate_Handle(t *testing.T) {
+	api, err := configureAPIServer()
+	if err != nil {
+		panic(err)
+	}
+
+	type want struct {
+		header     http.Header
+		statusCode int
+	}
+
+	tests := []struct {
+		name string
+		body string
+		want want
+	}{
+		{"passing", `{"Nickname": "toto", "Password": "1234"}`, want{header: http.Header{"Content-Type": {"application/json"}}, statusCode: 200}},
+		{"without Password", `{"Nickname": "toto"}`, want{header: http.Header{"Content-Type": {"application/json"}}, statusCode: 422}},
+		{"without Nickname", `{"Password": "1234"}`, want{header: http.Header{"Content-Type": {"application/json"}}, statusCode: 422}},
+		{"without Password and Nickname", `{}`, want{header: http.Header{"Content-Type": {"application/json"}}, statusCode: 422}},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
You should create a real function (not an anonymous one). That way, you will be able to reuse this function to create a new wallet before delete test.